File: imforensics/aggregator/aggregator.py
from __future__ import absolute_import

import os.path as op

class Aggregator(object):

    def __init__(self, matlab_engine):
        self.matlab_engine = matlab_engine
        ml_path = op.join(op.dirname(op.realpath(__file__)), 'matlab')
        self.matlab_engine.addpath(self.matlab_engine.genpath(ml_path))

    def aggregate(self, X_cm, X_ela, X_ho):
        return self.matlab_engine.aggregator(X_cm, X_ela, X_ho)

File: imforensics/aggregator/matlab/aggregator.m
function [X] = aggregator(X_cm, X_ela, X_ho)
    BLOCK_SIZE = 16;

    % Higer order stats
    X = X_ho;

    % Copy move
    X = cat(3, X, downsample(X_cm(:,:,1), BLOCK_SIZE));

    % ELA
    X = cat(3, X, downsample(X_ela(:,:,1), BLOCK_SIZE));
    X = cat(3, X, downsample(X_ela(:,:,2), BLOCK_SIZE));
    X = cat(3, X, downsample(X_ela(:,:,3), BLOCK_SIZE));
return
